### PR TITLE
[WIP] Etcd check

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

This addresses the kubeadm 1.9 to 1.10 issue for tls-enabled etcd.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/kubeadm/app/phases/upgrade/staticpods.go [hunk 1]
```diff
@@ -182,19 +183,38 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
...
-	// API endpoint below of the OLD Static Pod component and proceeds quickly enough, which might lead to unexpected results.
-	if err := waiter.WaitForStaticPodControlPlaneHashChange(cfg.NodeName, component, beforePodHash); err != nil {
-		return rollbackOldManifests(recoverManifests, err, pathMgr, recoverEtcd)
-	}
+	if component == constants.Etcd {
+		// Because we may be updating the tls settings for etcd during the upgrade, we cannot use the api server to verify the deployment
+		// instead poll for the etcd version after a short delay
+		etcdCluster := util.LocalEtcdCluster{
+			TLS:           true,
+			CertFile:      "/etc/kubernetes/pki/etcd/peer.crt",
```
Comment (timothysc) [inline cmd/kubeadm/app/phases/upgrade/staticpods.go]: These are all config params that should be referenced.

**Review by timothysc — CHANGES_REQUESTED** [review]
This is more in line with a minimal back-patch that I was looking for but we need to test the heck out of it.  

There are minor things that need to be changed such as referencing the config.

**Comment by timothysc** [comment]
@detiber @stealthybox - Can we merge this into a single PR and close this one please?

**Comment by stealthybox** [comment]
Moved to collab PR: https://github.com/stealthybox/kubernetes/pull/1

**Pull request closed without merging**